test/Microsoft.TestPlatform.CoreUtilities.UnitTests/Helpers/DotnetHostHelperTest.cs

```diff
@@ -343,6 +343,92 @@ public void GetDotnetPathByArchitecture_SameArchitecture()
         Assert.AreEqual(found ? expectedMuxerPath : null, muxerPath);
     }
 
+    [DataTestMethod]
+    [DataRow(DotnetMuxerResolutionStrategy.DotnetRootArchitecture)]
+    [DataRow(DotnetMuxerResolutionStrategy.DotnetRootArchitectureLess)]
+    [DataRow(DotnetMuxerResolutionStrategy.GlobalInstallationLocation)]
+    [DataRow(DotnetMuxerResolutionStrategy.DefaultInstallationLocation)]
+    [DataRow(DotnetMuxerResolutionStrategy.Default)]
+    public void GetDotnetPathByArchitecture_Strategies(DotnetMuxerResolutionStrategy strategy)
```

Comment:
Because you are switching on the strategy for the assertions, I would split the test into multiple tests (one for each strategy).